src/tiled/objecttypesmodel.cpp

```diff
@@ -27,6 +27,7 @@ void ObjectTypesModel::setObjectTypes(const ObjectTypes &objectTypes)
 {
     beginResetModel();
     mObjectTypes = objectTypes;
+    sort(0, Qt::AscendingOrder);
```

Comment:
I would probably just do this here instead:
```c++
qSort(mObjectTypes.begin(), mObjectTypes.end(), objectTypeLessThan);
```
Since there is no need for the additional signals about the layout.